web/src/templates.ts

```diff
@@ -173,3 +175,13 @@ Handlebars.registerHelper("popover_hotkey_hints", (...args) => {
         `<span class="popover-menu-hotkey-hints">${hotkey_hints}</span>`,
     );
 });
+
+// The below section is for registering global Handlebar partials.
+
+// The "input" partial block located at web/templates/components/input.hbs
+// is used to wrap any input element that needs to be styled as a Zulip input.
+// Usage example:
+//    {{#> input . input_type="filter-input" custom_classes="inbox-search-wrapper" icon="search" input_button_icon="close"}}
+//        <input type="text" id="{{INBOX_SEARCH_ID}}" class="input-element" value="{{search_val}}" autocomplete="off" placeholder="{{t 'Filter' }}" />
+//    {{/input}}
+Handlebars.registerPartial("input", render_input);
```

Comment:
Why use this partial rather than just the syntax `{{> ../components/action_button` we use for other components?

We should think of custom code in this file as very expensive from a maintenance standpoint, since you have to know a lot about the system to know to look here. It seems like this just saves a path characters in just 8 files.

Reply:
Removing the need to use the relative path was not the reason for this approach. I actually couldn't get the Handlebars [Partial Blocks](https://handlebarsjs.com/guide/partials.html#partial-blocks) to work with relative paths. The webpack build fails every time I try to use a relative path instead of registering the partial.

Reply:
I'd be happy to use the relative paths to refer the partial like in the case of action buttons, if there exists such a solution.

Reply:
This should definitely work, maybe someone else can pick it up though. Tracked in #35135.